### PR TITLE
CI: some time there are timeouts on DROP TABLES for random tests

### PR DESCRIPTION
**Pull request by CheSema**
I noticed that time to times a bunch of tests are getting timeouts.
I checked that it happens on DROP TABLE request at their end. 
In that time DatabaseCatalog reports that there are around 10K table in the queue.
Also I see that almost all that tables came from the 01526_client_start_and_exit.sh.

01526_client_start_and_exit.sh creates 10K tables and do not drop them explicitly. This tables are dropped along the database.

I made:
- mark that test as no-parallel
- remove the tables inside the test explicitly.

UPD: we just deleted 01526_client_start_and_exit.sh.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 803b58cdfa4000622e7f712541c87be006e015d7 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65535/803b58cdfa4000622e7f712541c87be006e015d7/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Cloud fork sync (only for ClickHouse Inc. employees)</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/01526_client_start_and_exit.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env bash
-# Tags: no-fasttest
+# Tags: no-fasttest, no-parallel
```
Comment (Algunenano) [inline tests/queries/0_stateless/01526_client_start_and_exit.sh:2]: I would be ok if we remove or disable this test. It's doing a crazy things to the server to check a basic/corner case bug in the client. It's not super useful.

WDYT?

Comment (CheSema) [reply tests/queries/0_stateless/01526_client_start_and_exit.sh:2]: I don't have strong opinion about that tests. I could only say that makeing 10K tables to check something in the client is not optimal. I do not understand why is there 10K instead 10?

**Comment by CheSema** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/64478/19bafb5063ece9d71f44f248b35087800c4f302c/stateless_tests__msan__[6_6].html

This is the example when some unrelated tests are being timeouted.

File: tests/queries/0_stateless/01526_client_start_and_exit.sh (hunk 1 above)
Comment (Algunenano) [reply tests/queries/0_stateless/01526_client_start_and_exit.sh:2]: I would just remove it. We are trying to get rid of slow no-parallel tests too, so I would not bother.

Comment (CheSema) [reply tests/queries/0_stateless/01526_client_start_and_exit.sh:2]: Then lets remove it in that PR with all that context. I will do that.

**Review by Algunenano — APPROVED** [review]

**Pull request merged**